scipy/stats/tests/test_distributions.py

```diff
@@ -9646,6 +9646,53 @@ def test_sf(self):
         assert_allclose(sf1, sf0)
 
 
+def test_irwinhall():
+    unif = stats.uniform(0, 1)
+    ih1 = stats.irwinhall(1)
+
+    pts = np.linspace(0, 1, 100)
+    pdf_unif = unif.pdf(pts)
+    pdf_ih1 = ih1.pdf(pts)
+    assert_allclose(pdf_ih1, pdf_unif) # should be identical to uniform
+
+    cdf_unif = unif.cdf(pts)
+    cdf_ih1 = ih1.cdf(pts)
+    assert_allclose(cdf_ih1, cdf_unif) # should be identical to uniform
+
+    ih2 = stats.irwinhall(2)
+    pdf_ih2 = ih2.pdf(1)
+    assert_allclose(pdf_ih2, 1) # should be 1 at x=1
+
+    npts = 100
+    pts = np.linspace(0, 2, npts)
+    expected = np.linspace(0, 2, npts)
+    expected[npts//2:] = 2 - expected[npts//2:]
+    pdf_ih2 = ih2.pdf(pts)
+    assert_allclose(pdf_ih2, expected) # should be a triangle
+
+    # PDF is symmetric so CDF should be 0.5 at n/2
+    for n in range(2, 10):
+        ih = stats.irwinhall(n)
+        ih_cdf = ih.cdf(n/2)
+        assert_allclose(ih_cdf, 0.5)
+
+    ih10 = stats.irwinhall(10)
+    # from Wolfram Alpha: "values of pdf of UniformSumDistribution[10] for x=0 to x=10"
+    # symmetric about n/2 = 5
+    vals = [0, 1/362880, 251/181440, 913/22680, 44117/181440]
+    vals += [15619/36288] + vals[::-1]
+    assert_allclose(ih10.pdf(np.arange(11)), vals)
+
+    # check that the definition of the Irwin-Hall distribution is correct
+    nsamp = 1000
+    uu = unif.rvs(size=nsamp, random_state=1234)
+    sum10s = np.bincount(np.arange(len(uu))//10, uu)
+    res = stats.kstest(sum10s, ih10.cdf)
+    assert res.pvalue > 0.1
+
+
+
+
```

Comment:
I see. So even if they currently execute pretty much the same test, the _intent_ of this one is to be a weak test of the CDF implementation, whereas the intent of `check_distribution_rvs` is to be a test of consistency between the RVS and CDF.

In any case, you have a much stronger test of the CDF above (i.e. comparing actual values against Wolfram Alpha). So unless the previous test fails, I can only see this test producing false positives. You don't need to remove it; I can do that too.

Reply:
The idea was to check the distributional relation between IH(N) and summing N iid uniforms. As far as I can tell the other checks test the random variable generation against its own cdf implementation vs the cdf of a different distribution.

Practically they should always pass since we do the sum of uniforms way to make IH samples, but suppose someone changed that to use the inverse of the cdf or some other generation algorithm (ziggurat?) instead. 

We'd have to make sure that summing N uniforms and taking a single IH(N) sample are distributionally identical or close to it.

I guess it's like testing that chi-square is really the distribution of a squared normal r.v. This should hold regardless of how normal and chi-square are implemented.

Reply:
Well, to [quote Knuth](https://staff.fnwi.uva.nl/p.vanemdeboas/knuthnote.pdf) on this: 
>Beware of bugs in the above code; I have only proved it correct, not tried it.

I'll remove it for now